### PR TITLE
Transform key hashable

### PR DESCRIPTION
**Pull request by coltfred**
- To solve our immediate problem I owned up to the fact that we were already exposing public apis with Hashable in them and exposed the Hashable instance for TransformKey.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -399,6 +400,12 @@ pub struct EncryptedTempKey {
     _internal_fp12: Fp12Elem<Fp256>,
 }
 
+impl Hashable for EncryptedTempKey {
+    fn to_bytes(&self) -> ByteVector {
+        self._internal_fp12.to_bytes()
```
Comment (clintfred) [inline src/api.rs]: Would it make sense to just use `.bytes()` here?
```suggestion
        self.bytes().to_vec()
```

File: src/api.rs [hunk 2]
```diff
@@ -437,6 +444,12 @@ pub struct HashedValue {
     _internal_value: HomogeneousPoint<Fp2Elem<Fp256>>,
 }
 
+impl Hashable for HashedValue {
+    fn to_bytes(&self) -> ByteVector {
+        self._internal_value.to_bytes()
```
Comment (clintfred) [inline src/api.rs]:
```suggestion
        self.bytes().to_vec()
```

File: src/api.rs [hunk 3]
```diff
@@ -501,7 +514,20 @@ pub struct TransformKey {
     hashed_temp_key: HashedValue,
     public_signing_key: PublicSigningKey,
     signature: Ed25519Signature,
-    _internal_key: internal::SignedValue<internal::ReencryptionKey<Fp256>>,
+    pub(crate) _internal_key: internal::SignedValue<internal::ReencryptionKey<Fp256>>,
```
Comment (clintfred) [inline src/api.rs]: Does this need to be `pub(crate)`?
```suggestion
    _internal_key: internal::SignedValue<internal::ReencryptionKey<Fp256>>,
```

Comment (coltfred) [reply src/api.rs]: I thought it had to be so we could use it from the impl, but I was wrong. Thanks!

**Comment by coltfred** [comment]
@clintfred Until we decide that serde would be a better role overall I'm hesitant to switch to it. It seems like switching would be a good idea, but we'd need to prove it out before I'd want to integrate it into the public api.

**Pull request merged**